pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java

```diff
@@ -18,50 +18,32 @@
 import com.linkedin.pinot.core.data.GenericRow;
 
 
+/**
+ * Interface for a consumer that consumes at stream level and is unaware of any partitions of the stream
+ */
 public interface StreamLevelConsumer {
 
   /**
-   *
+   * Initialize and start the stream level consumer
+   * @throws Exception
    */
   void start() throws Exception;
 
   /**
-   *
-   * @param offset
-   */
-  void setOffset(long offset);
-
-  /**
-   * return GenericRow
-   */
-  GenericRow next(GenericRow destination);
-
-  /**
-   *
-   * @param offset
+   * Get next row from the stream and decode it into a generic row
+   * @param destination
    * @return
    */
-  GenericRow next(long offset);
-
-  /**
-   *
-   * @return
-   */
-  long currentOffset();
+  GenericRow next(GenericRow destination);
 
   /**
-   *
+   * Commit the offsets
```

Comment:
Please add a comment here that the next call to consume should exclude all events consumed until the commit() method is called, and start from newer events not yet consumed